c10/core/InferenceMode.cpp

```diff
@@ -0,0 +1,10 @@
+
+#include <c10/core/InferenceMode.h>
+
+namespace c10 {
+
+bool InferenceMode::is_enabled() {
+  return !c10::impl::tls_is_dispatch_key_included(DispatchKey::InplaceOrView);
```

Comment:
If we're planning on calling this function from perf critical code (which I think we do since we call this in every TensorImpl creation), I would suggest we actually save the boolean argument given to the InferenceMode constructor and just return that here. That will avoid the TLS read.

Reply:
Oh actually I think we cannot avoid the TLS read here. since this function is static that we call with `InferenceMode::is_enabled()`, it's not associated to any InferenceMode object. If we saved some boolean during constructor, it'll have to be a TLS boolean as well which makes no difference in perf.

Reply:
Ho it is static, my bad.
TLS read it is then :)